zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/task/JobWorkerTaskProcessor.java

```diff
@@ -95,7 +95,6 @@ public Class<ExecutableJobWorkerTask> getType() {
   public void onTerminate(final ExecutableJobWorkerTask element, final BpmnElementContext context) {
     final var flowScopeInstance = stateBehavior.getFlowScopeInstance(context);
 
-    jobBehavior.cancelJob(context);
```

Comment:
:wrench: We could keep the job canceling to be explicit about the behavior (here and in the other two processors). Otherwise, it may be confusing how jobs are canceled if you inspect the processor. 

For event subscriptions and incidents, we call the behavior explicitly in each processor. We should be consistent.